microsoft/PowerToys
# Adding query caching and delayedExecution to PT Run OneNote plugin

## Summary of the Pull Request


## PR Checklist

- [x] Closes #18846 
- [N/A] **Tests:** Added/updated and all pass
- [x] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: https://github.com/MicrosoftDocs/windows-uwp/pull/3893


## Detailed Description of the Pull Request / Additional comments

Adds caching to the plugin results for 1 day per query, using the LazyCache library. Returns cached results in initial Query() call, otherwise queries OneNote itself in the delayedExecution:true call.


## Validation Steps Performed

(with win32 OneNote installed) Run PowerLauncher, and in the dialog type "o:", then press enter on a result to open it in OneNote. Or you can run the runner, and then enable the OneNote plugin in the UI and do the above. If you query OneNote, the first time there will be some lag (~1s) since it's querying OneNote, but if you run the same query a second time, the results should be virtually instantaneous
